apis/vshn/v1/common_types.go

```diff
@@ -208,3 +212,31 @@ type VSHNAccess struct {
 	// created in the namespace where the claim is located.
 	WriteConnectionSecretToReference *xpv1.SecretReference `json:"writeConnectionSecretToRef,omitempty"`
 }
+
+func (a *TimeOfDay) GetTime() time.Time {
+	v, err := time.Parse(time.TimeOnly, string(*a))
+	if err != nil {
```

Comment:
Instead of getting an error I thought it's better to set the time to 00:00:00. I could also try to return a nil time? what do you think

Reply:
I'm OK with such approach, but can You please write tests for those fuctions? Working with time is always tricky and I would like to see some tests in here for future

Reply:
Either return an error and let the caller side decide what to do with it or at least document it. 

Also this will fail on `nil` TimeOfDay objects. Either fix the signature to `func (a TimeOfDay) GetTime()`, which is fine because it does not modify `a` or add a nil check.

Reply:
> Either return an error and let the caller side decide what to do with it or document it. Returning zero time is ok (if documented) since the caller can use `time.Time.IsZero()`.

Thanks for input

> Also this will fail on `nil` TimeOfDay objects. Either fix the signature to `func (a TimeOfDay) GetTime()`, which is fine because it does not modify `a` or add a nil check.

Unfortunately I change the it [here](https://github.com/vshn/appcat/pull/195/files/0be649d04b7edefcedcc5d73870888cace0bf853#diff-e4106c7dc200d769b1c8a5341f5b5bb8d4cd00ed738e444a0d19ea04a6932923R235)

Reply:
> I'm OK with such approach, but can You please write tests for those fuctions? Working with time is always tricky and I would like to see some tests in here for future

Added unit tests and also checking for nil.

Reply:
> You can still change the signature of the GetTime function though

I didn't changed the signature because I am saving the state therefore I can only use a pointer. Nevertheless I am checking for nil pointers now and returning time zero.

Reply:
> I am saving the state therefore I can only use a pointer

You're not saving state in the getter. So you don't need a pointer receiver there. It's very much save to mix and match. In fact the standard library does that https://pkg.go.dev/time#Time

But plus one for `nil` check. This allow ppl to call GetTime() on a typed nil.